olp-cpp-sdk-core/tests/http/unittest/TestNetwork.cpp

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include <gtest/gtest.h>
+#include <olp/core/http/HttpStatusCode.h>
+#include <olp/core/http/Network.h>
+#include <olp/core/logging/Log.h>
+
+#include <algorithm>
+#include <atomic>
+#include <condition_variable>
+
+TEST(TestNetwork, get_request) {
+  auto network = olp::http::CreateDefaultNetwork(1);
+
+  auto payload = std::make_shared<std::stringstream>();
+
+  std::mutex m;
+  std::condition_variable cv;
+
+  olp::http::NetworkRequest request("http://localhost:3000/get_request");
```

Comment:
We should use proxy settings for this and let the SDK call normal URL so that we can react based on those.

Reply:
added also a test for HTTP proxy with basic auth.

Reply:
So these are only for testing Network layer or clients?

Reply:
Network at the moment